lib/trollop.rb

```diff
@@ -47,6 +244,56 @@ class VersionNeeded < StandardError
 ## and consider calling it from within
 ## Trollop::with_standard_exception_handling.
 class Parser
+
+  ## The registry is a class-variable Hash of registered option types.
+  ## By default, it contains flag(boolean), int, float, string, io and date types
+  ## and their plural versions.
+  @registry = {
+    # single opts
+    :fixnum => IntegerOption,
+    :int => IntegerOption,
+    :integer => IntegerOption,
+    :float => FloatOption,
+    :double => FloatOption,
+    :string => StringOption,
+    :bool => BooleanOption,
+    :boolean => BooleanOption,
+    :flag => BooleanOption,
+    :io => IOOption,
+    :date => DateOption,
+    :trueclass => BooleanOption,
+    :falseclass => BooleanOption,
+    ## array-arg opts
+    :fixnums => IntegerArrayOption,
+    :ints => IntegerArrayOption,
+    :integers => IntegerArrayOption,
+    :doubles => FloatArrayOption,
+    :floats => FloatArrayOption,
+    :strings => StringArrayOption,
+    :dates => DateArrayOption,
+    :ios => IOArrayOption,
+  }
+  
+  ## Gets the class from the registry.
+  ## Can be given either a class-name, e.g. Integer, a string, e.g "integer", or a symbol, e.g :integer
+  def self.registry_getklass(type)
+    return nil unless type
+    if type.respond_to?(:name)
```

Comment:
Is it possible to simplify this interface. Do we need to support so many `type` options?

Reply:
This is for backward compatibility with all the types Trollop supports, as you can do stuff like:
```ruby
opt :pig, "a pig", :type => :string
opt :wig, "a wig", :type => String
opt :fig, "a fig", :type => TrueClass
opt :big, "make it big", :type => :bool
```
Another option would be a case statement.  Is there a better way?

Reply:
Keep thanks